openedx_proversity_reports/reports/backend/enrollment_per_site_report.py

```diff
@@ -81,24 +88,79 @@ def generate_enrollment_per_site_report(course_key, enrolled_users):
             role: User's course role.
         }]
     """
-    opaque_course_key = CourseKey.from_string(course_key)
     report_data = []
+    bigquery_data = list(
+        get_google_bigquery_data(
+            query_string=get_google_bigquery_query(
+                course_dataset_name=get_google_bigquery_course_id(course_key),
+                date=date,
+                course_id=str(course_key),
+            ),
+        ),
+    )
 
     for user in enrolled_users:
         enrollment = get_course_enrollment().objects.filter(
             user__email=user.get('email', ''),
-            course_id=opaque_course_key,
+            course_id=course_key,
         )
 
         if not enrollment:
             continue
 
+        time_spent_per_user = 0
+
+        try:
```

Comment:
What if we do the following:

```suggestion
        try:
            time_spent_per_user = sum(row_data[1] for row_data in bigquery_data if row_data[0] == user.get('username', ''))
        except IndexError:
            pass
```

Reply:
`sum()` is already optimized to handle the operation.  @Squirrel18